proposals/0003-cocoapods-support-improvements.md

```diff
@@ -0,0 +1,117 @@
+---
+title: CocoaPods Support Improvements
+author:
+- Orta Therox
+- Eloy Durán
+date: 2018-08-15
+---
+
+# RFC0003: CocoaPods Support Improvements
+
+## Summary
+
+Simplify CocoaPods support in React Native, by reducing the differences between how CocoaPods maps React Native into the existing set of Xcode projects.
+
+## Basic example
+
+CocoaPods support in React Native is a tad tricky, because it’s trying to map a complex set of Xcode projects into a single dependency. The Podspec for React Native does a lot of work under the hood at ~300 LOC. This is, in part, because we use a CocoaPods abstraction called subspecs to consolidate out the React Native library into one single library.
+
+If we split the `React.podspec` into many podspecs, one per Xcode project. Roughly ~10ish Podspecs per React Native release.  These would match up one-to-one to the Xcode projects so that when CocoaPods support is broken in React Native, an error report would make it obvious which part of the project is broken.
+
+This means we can also move React Native and we can get CocoaPods support back in CI. This route allows CocoaPods support to be tested without network access (all the required information would be inside the same)
+
+The CocoaPods React dependency would then depend on React-ART, and React-RCTSettings etc.
+
+## Motivation
+
+For teams adding React Native to existing apps CocoaPods support let's RN be treated as a dependency like all of the other ones. CocoaPods is built with a lot of escape valves for consumers to fix issues at runtime (you see people passing around `post_install` [fixes](https://github.com/facebook/react-native/issues/20182#issuecomment-409131862) for example, and I built a [CocoaPods plugin](https://github.com/orta/cocoapods-fix-react-native) to try consolidate them all)
+
+I think one of the a key reasons for the drift between the CocoaPods support and the Xcode projects is because they're not mapped together.
```

Comment:
What implications does this have to the existing consumers of the Xcode projects? My understanding is that they'll be unused by CocoaPods. How will this ensure that CocoaPods and the existing Xcode projects stay in sync?

Reply:
That is to say, we've had [bugs such as this](https://github.com/facebook/react-native/issues/19569) which were a result of an Xcode project being misconfigured. How will changing this CocoaPods structure allow for dependency changes like these to be addressed?

Reply:
I believe man-made Xcode projects in the repo must die, they must be build artifacts of Buck packages. Nobody should configure builds in more than one place.